src/main/java/de/retest/recheck/TestCaseFinder.java

```diff
@@ -86,7 +88,41 @@ public static TestCaseFinder getInstance() {
 	 * @return The class name for the test case method in all stack traces.
 	 */
 	public Optional<String> findTestCaseClassNameInStack() {
-		return findTestCaseMethodInStack( toClassName() );
+		final Optional<String> testCaseClassName = findTestCaseMethodInStack( toClassName() );
+		if ( testCaseClassName.isPresent() ) {
+			return testCaseClassName;
+		} else {
+			return findTestCaseClassInStack();
+		}
+	}
+
+	/**
+	 * @return The class name for the test being called from.
+	 */
+	public Optional<String> findTestCaseClassInStack() {
+		for ( final StackTraceElement[] stack : Thread.getAllStackTraces().values() ) {
```

Comment:
This removes the need for two methods:
```
return Thread.getAllStackTraces().values().stream() //
				.flatMap( Stream::of ) //
				.map( StackTraceElement::getClassName ) //
				.filter( testClassSuffixes::contains ) //
				.findFirst();
```

Reply:
Pascal already incorporated a stream-based approach. However, we decided to have public variants with and without passing a stack trace, mostly for being consistent with the remaining API.